sdks/java/extensions/zetasketch/src/test/java/org/apache/beam/sdk/extensions/zetasketch/ApproximateCountDistinctTest.java

```diff
@@ -0,0 +1,342 @@
+
+    PAssert.that(result)
+        .containsInAnyOrder(
+            ImmutableList.of(
+                KV.of(0, INTS1_ESTIMATE), KV.of(1, INTS1_ESTIMATE), KV.of(2, INTS1_ESTIMATE)));
+
+    p.run();
+  }
+
+  /** Test String Globally. */
+  @Test
+  @Category(NeedsRunner.class)
+  public void testStandardTypesPerKeyForStrings() {
+    List<KV<Integer, String>> strings = new ArrayList<>();
+
+    for (int i = 0; i < 3; i++) {
+      for (int k : INTS1) {
+        strings.add(KV.of(i, String.valueOf(k)));
+      }
+    }
+
+    PCollection<KV<Integer, Long>> result =
+        p.apply("Str", Create.of(strings)).apply("StrHLL", ApproximateCountDistinct.perKey());
+
+    PAssert.that(result)
+        .containsInAnyOrder(
+            ImmutableList.of(
+                KV.of(0, INTS1_ESTIMATE), KV.of(1, INTS1_ESTIMATE), KV.of(2, INTS1_ESTIMATE)));
+
+    p.run();
+  }
+
+  /** Test Byte Globally. */
+  @Test
+  @Category(NeedsRunner.class)
+  public void testStandardTypesPerKeyForBytes() {
+
+    List<KV<Integer, byte[]>> bytes = new ArrayList<>();
+
+    for (int i = 0; i < 3; i++) {
+      for (int k : INTS1) {
+        bytes.add(KV.of(i, ByteBuffer.allocate(4).putInt(k).array()));
+      }
+    }
+
+    PCollection<KV<Integer, Long>> result =
+        p.apply("BytesHLL", Create.of(bytes)).apply(ApproximateCountDistinct.perKey());
+
+    PAssert.that(result)
+        .containsInAnyOrder(
+            ImmutableList.of(
+                KV.of(0, INTS1_ESTIMATE), KV.of(1, INTS1_ESTIMATE), KV.of(2, INTS1_ESTIMATE)));
+
+    p.run();
+  }
+
+  /** Test a general object, we will make use of a KV as the object as it already has a coder. */
+  @Test
+  @Category(NeedsRunner.class)
+  public void testObjectTypesGlobal() {
+
+    PCollection<Long> approxResultInteger =
+        p.apply(
+                "Int",
+                Create.of(
+                    INTS1.stream().map(x -> KV.of(x, KV.of(x, x))).collect(Collectors.toList())))
+            .apply(
+                "IntHLL",
+                ApproximateCountDistinct.<KV<Integer, KV<Integer, Integer>>>globally()
+                    .via((KV<Integer, KV<Integer, Integer>> x) -> (long) x.getValue().hashCode()));
+
+    PAssert.thatSingleton(approxResultInteger).isEqualTo(INTS1_ESTIMATE);
+
+    p.run();
+  }
+
+  /** Test a general object, we will make use of a KV as the object as it already has a coder. */
+  @Test
+  @Category(NeedsRunner.class)
+  public void testObjectTypesPerKey() {
+
+    List<KV<Integer, KV<Integer, Integer>>> ints = new ArrayList<>();
+
+    for (int i = 0; i < 3; i++) {
+      for (int k : INTS1) {
+        ints.add(KV.of(i, KV.of(i, k)));
+      }
+    }
+
+    PCollection<KV<Integer, Long>> approxResultInteger =
+        p.apply("Int", Create.of(ints))
+            .apply(
+                "IntHLL",
+                ApproximateCountDistinct.<Integer, KV<Integer, Integer>>perKey()
+                    .via(x -> KV.of(x.getKey(), (long) x.hashCode()))
```

Comment:
Same here. Shouldn't we use `x -> KV.of(x.getKey(), x.getValue().hashCode())` here?